File: CsharpProjects/StrategyExemplo04/StrategyExemplo04/Pedidos/Pedido.cs
﻿using StrategyExemplo04.Fretes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace StrategyExemplo04.Pedidos
{
    public abstract class Pedido
    {
        public double valor { get; set; }
        public string? nomeSetor { get; set; }
        public Frete tipoFrete { get; set; }

        public double calculaFrete()
        {
            return tipoFrete.calcula(this.valor);
        }
    }
}

File: JavaProjects/StrategyExemplo04/src/Pedidos/Pedido.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package Pedidos;

import frete.Frete;

/**
 *
 * @author jeffe
 */
public abstract class Pedido {
    
    private double valor;
    private String nomeSetor;
    private Frete tipoFrete;

    public double getValor() {
        return valor;
    }

    public void setValor(double valor) {
        this.valor = valor;
    }

    public String getNomeSetor() {
        return nomeSetor;
    }

    public void setNomeSetor(String nomeSetor) {
        this.nomeSetor = nomeSetor;
    }

    public Frete getTipoFrete() {
        return tipoFrete;
    }

    public void setTipoFrete(Frete tipoFrete) {
        this.tipoFrete = tipoFrete;
    }
    
    public double calculaFrete(){
        return tipoFrete.calcula(valor);
    }
    
}
